mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -336,4 +339,36 @@
      */
     List<MailboxPath> list(MailboxSession session) throws MailboxException;
 
-}
+    /**
+     * Return all mailbox's annotation as the {@link List} of {@link MailboxAnnotation}
+     * 
+     * @param mailboxPath   the current mailbox
+     * @param session       the current session
+     * @return              List<MailboxAnnotation>
+     * @throws MailboxException
+     */
+    List<MailboxAnnotation> getAnnotations(MailboxPath mailboxPath, MailboxSession session) throws MailboxException;
+
+    /**
+     * Get all mailbox's annotation by the list of the keys
```

Comment:
I used the overriding methods with the paramter key as the filter value instead of point on the method name.
If not, we can name the method like getAllAnnotations and getAnnotationsByKeys
Which one do we prefer?